typescript/helloworld/src/app/app.ts

```diff
@@ -1,26 +1,40 @@
 import { BigNumber, ethers } from 'ethers';
 
-import { TypedListener } from '@abacus-network/core/dist/common';
 import {
   AbacusApp,
+  AbacusCore,
+  ChainMap,
   ChainName,
   ChainNameToDomainId,
+  MultiProvider,
   Remotes,
 } from '@abacus-network/sdk';
-
-import { ReceivedHelloWorldEvent } from '../types/contracts/HelloWorld';
+import { utils } from '@abacus-network/utils';
 
 import { HelloWorldContracts } from './contracts';
 
+type Counts = {
+  sent: number;
+  received: number;
+};
+
 export class HelloWorldApp<
   Chain extends ChainName = ChainName,
 > extends AbacusApp<HelloWorldContracts, Chain> {
+  constructor(
+    public readonly core: AbacusCore<Chain>,
+    contractsMap: ChainMap<Chain, HelloWorldContracts>,
+    multiProvider: MultiProvider<Chain>,
+  ) {
+    super(contractsMap, multiProvider);
+  }
+
   async sendHelloWorld<From extends Chain>(
     from: From,
     to: Remotes<Chain, From>,
     message: string,
     value: BigNumber,
-    receiveHandler?: TypedListener<ReceivedHelloWorldEvent>,
+    timeoutMs?: number,
```

Comment:
So how it is setup right now, is if there is a timeout specified it will wait for that time OR until the send is done, whichever happens first. And without the timeout it just waits for the send to be done. The alternative I see is to just have the caller be responsible for the timeout which makes more sense anyway now that we have it as a util.

Reply:
Moved